### ISSUE TITLE
Centering the input data is not equivalent to removing the linear trend beforehand

### ISSUE THREAD

**Issue by markusritschel** [nicrie/xeofs#108]
Dear Niclas,

First of all, thanks a lot for this wonderful package. It's nice to watch it grow, providing more and more functionalities over time :-)

I do have one question/concern about the `center` parameter in the `EOF` model.
I would assume that setting `center=True` is equivalent to removing the linear trend or (in case of no trend) the long-term average beforehand. Such that the following two approaches result in (at least nearly) the same.
```python
import xarray as xr
from xarrayutils import xr_detrend
from xeofs.models import EOF

da = xr.tutorial.load_dataset('air_temperature').air

# 1st approach
m1 = EOF(n_modes=20, standardize=False, center=True)
m1.fit(da, dim=['lon', 'lat'])

# 2nd approach
da = xr_detrend(da, 'time')
# da = da - da.mean('time')
m2 = EOF(n_modes=20, standardize=False, center=False)
m2.fit(da, dim=['lon', 'lat'])
```
However, this is not the case as can be seen when we look at the different EOFs or explained variance etc.
Or do I misinterpret something here? If so, feel free to correct me :-)

**Desktop:**
OS: Ubuntu
xeofs version: 1.2.0

**Comment by nicrie** [comment]
Hi Markus! Thanks for the kind words! The `center` parameter only removes the mean of the data but not the linear trend. Probably the term centering is used differently in some communities, sorry for the confusion.  

There is currently no direct function in xeofs for removing global trends in the data. 

I don't know if this is relevant for you, but you could try using [Extended EOF](https://xeofs.readthedocs.io/en/latest/_autosummary/xeofs.models.ExtendedEOF.html) /Multi-Channel Singular Spectrum Analysis to remove global non-linear trends in your data though :)

**Comment by markusritschel** [comment]
Hi Niclas,

Thanks for the quick response! 
Ok. After looking more closely at the code again, I guess (in case of S-mode EOF analysis, i.e. EOF or scores = spatial pattern, components = temporal evolution) [the routine removes the field average for every single time step (`sample_dims`)](https://github.com/nicrie/xeofs/blob/1f38a5b818a5d5fc55720e1119e6f686aef89168/xeofs/preprocessing/scaler.py#L102C15-L102C15) and not the long-term mean, right? Maybe this can be formulated more clearly in the documentation.?
Because I assumed you subtract the long-term mean in every grid cell by setting `center=True`.

**Comment by nicrie** [comment]
Ahh, perhaps there is a small mix-up here?

[Compagnucci & Richman (2007)](https://d1wqtxts1xzle7.cloudfront.net/7362406/ijc-compagnuccirichman-libre.pdf?1390851172=&response-content-disposition=inline%3B+filename%3DCan_principal_component_analysis_provide.pdf&Expires=1699465656&Signature=ZJ2T5bEvdGau3wjBpsFETVcSoUmJ9vrEQJMPasMW7gcFIj39nNdP8BGcQehggsRx01Zj-enTEleDqhI0hEcwrzzRq~4lNUmDpiCwMMA3msHkfTe3yLsxMDbRSSSw8mBlG9n-F0SwGBNr1cyfaLSJP5lNTTGWig6c7gAADrQ4hPMaSnfqXpu8vumjRygHmp8x03080nYqwmteEBKtVnWGEQuDIPjfS5ZfW9OwxEnjnFpVIwZDWxoGGKDyyrZJF5W3bkPL2JzpQZCKG2eEQv-Ja960V85QyP9hYfjIq-TBpFApB0keVhd40FR7-9JxYhbEz3VLqNjPLZ8CiD7tdvZNFw__&Key-Pair-Id=APKAJLOHF5GGSLRBV4ZA) define for m time steps at n stations (or grid points):

* **S-mode** treats the time series (m times) at each of the n stations (or grid points) as variables in the analysis; the domain is the geographical area.
* **T-mode** treats the spatial field, defined by all the n stations (or grid points), at each of the m times as variables; the domain is time.

And further,
> S-mode analysis results in an nxn similarity matrix whereas the T-mode will result in an mxm similarity matrix.

So if the scores are the spatial patterns and the components are the temporal evolution, then it is T-mode.

> Because I assumed you subtract the long-term mean in every grid cell by setting center=True.

Yes, this is true for S-mode, but not for T-mode

perhaps it's easiest to sum up like this:

| Analysis Mode | `dim` in xeofs | scores      | components | `center` in xeofs |
| ----------------------|-------------------|---------------|--------------------|---------------------------------------------------- |
| S-mode             | ("time")            | temporal | spatial           | remove temporal mean for every grid point |
| T-mode             | ("lon", "lat")   | spatial     | temporal       | remove spatial mean for every time step |

**Comment by markusritschel** [comment]
Ou, yes, ok, now it makes all sense 😅 
I guess, the mix-up arose with the new version I installed of your package.
I used v0.7.0 until recently, in which the _Principle Components_ were called _components_. 
In the new version, you introduced a new nomenclature for the _EOFs_ (eigenvectors; formerly _eofs_, now _components_) and the _Principle Components_ (formerly _components_, now _scores_).

Apparently, this led to confusion on my side. Thanks for the exchange and the clarification! 👍🏽
